src/store/sagas/setting.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { takeLatest, all, fork, call, put } from 'redux-saga/effects';
+
+import {
+  fetchApplianceCategorySuccess,
+  fetchApplianceCategoryError,
+  fetchUnitsSuccess,
+  fetchUnitsError,
+  fetchParametersSuccess,
+  fetchParametersError,
+} from 'store/actions/setting';
+import { errorHandler } from 'store/helpers';
+import api, { applianceCategoryRequest, parametersRequest, unitsRequest } from 'services/api';
+import { settingsConstants } from 'store/actions/setting';
+
+function* callFetchApplianceCategory(action) {
+  try {
+    const orgId = action.payload.params.orgId;
+    const url = applianceCategoryRequest.APPLIANCE_CATEGORY.replace(':orgId', orgId);
+    const { data } = yield call([api, 'get'], url, action.payload);
+    yield put(fetchApplianceCategorySuccess(data));
+  } catch (error) {
+    yield fork(errorHandler, error, fetchApplianceCategoryError);
+  }
+}
+
+function* callFetchUnits(action) {
+  try {
+    const orgId = action.payload.params.orgId;
+    const url = unitsRequest.UNITS.replace(':orgId', orgId);
+    const { data } = yield call([api, 'get'], url, action.payload);
+    yield put(fetchUnitsSuccess(data));
+  } catch (error) {
+    yield fork(errorHandler, error, fetchUnitsError);
+  }
+}
+
+function* callFetchParameters(action) {
+  try {
+    const orgId = action.payload.params.orgId;
+    const url = parametersRequest.PARAMETER.replace(':orgId', orgId);
+    const { data } = yield call([api, 'get'], url, action.payload);
+    yield put(fetchParametersSuccess(data));
+  } catch (error) {
+    yield fork(errorHandler, error, fetchParametersError);
+  }
+}
+
+export function* watchFetchUnits() {
+  yield takeLatest(settingsConstants.FETCH_UNITS_REQUEST, callFetchUnits);
+}
+
+export function* watchFetchApplianceCategory() {
+  yield takeLatest(settingsConstants.FETCH_APPLIANCE_CATEGORY_REQUEST, callFetchApplianceCategory);
+}
+
+export function* watchFetchParameters() {
+  yield takeLatest(settingsConstants.FETCH_PARAMETERS_REQUEST, callFetchParameters);
+}
+
+export default function* settingsSaga() {
+  yield all([fork(watchFetchApplianceCategory), fork(watchFetchUnits), fork(watchFetchParameters)]);
+}
```

Comment:
I think all these can be grouped

Reply:
Nice didn't know this was possible until now. Thanks for that